core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -353,17 +352,22 @@ TableScanContext planWith(ExecutorService executor) {
         toSnapshotId,
         executor,
         fromSnapshotInclusive,
-        metricsReporters);
+        metricsReporter);
   }
 
-  Collection<MetricsReporter> metricsReporters() {
-    return metricsReporters;
+  MetricsReporter metricsReporter() {
+    return metricsReporter;
```

Comment:
Rather than having a default that requires awkward logic in `reportWith`, why not set the default to `null` and return `LoggingMetricsReporter.instance()` here if the reporter is null? That seems easier to me.

Reply:
agreed, that seems better